Fix/Remove potentially confusing references to Euler

willvincent opened the pull request:
e is referred to as Euler’s number, so technically the MATH_EXP description in VisualScript doc was not incorrect, though could potentially lead to confusion.

e is different from Euler’s constant however, making the existing GDScript exp description invalid.

Fixed #18179

aaronfranke commented on modules/visual_script/doc_classes/VisualScriptBuiltinFunc.xml:
```diff
@@ -80,7 +80,7 @@
 		</constant>
 		<constant name="MATH_EXP" value="20" enum="BuiltinFunc">
-			Return [b]e[/b] raised to the power of the input. [b]e[/b] sometimes called "Euler's number" is a mathematical constant whose value is approximately 2.71828.
+			Return the mathematical constant [b]e[/b] raised to the power of the input. [b]e[/b] has an approximate value of 2.71828.
```
I know this is existing text, but I think it would be better as "raised to the specified power" rather than "raised to the power of the input".

willvincent replied:
Would probably need to adjust wording on other descriptions in the VisualScriptBuiltinFunc file for consistency, there are several instances of "of the input" wordings...

aaronfranke reviewed (approved):
I didn't find the existing text too confusing personally but I agree this version is better.

Noshyaar commented:
The file that contains incorrect info is VisualScriptMathFunc

Noshyaar commented on doc/classes/@GDScript.xml:
```diff
@@ -326,7 +326,8 @@
 			<description>
-				Raises the Euler's constant [b]e[/b] to the power of [code]s[/code] and returns it. [b]e[/b] has an approximate value of 2.71828.
+			  The natural exponential function. It raises the mathematical constant [b]e[/b] to the power of [code]s[/code] and returns it.
+				[b]e[/b] has an approximate value of 2.71828.
```
indentation issue?

willvincent replied:
Matches all the indents of everything else in the file on my screen...

Noshyaar commented on doc/classes/@GDScript.xml:
```diff
@@ -326,7 +326,8 @@
 			<description>
-				Raises the Euler's constant [b]e[/b] to the power of [code]s[/code] and returns it. [b]e[/b] has an approximate value of 2.71828.
+				The natural exponential function. It raises the mathematical constant [b]e[/b] to the power of [code]s[/code] and returns it.
+				[b]e[/b] has an approximate value of 2.71828.
```
This will create another paragraph. (One line is one paragraph)

If this is intended, then I'm fine with it. Just letting you know why it was on the same line before. :)

Noshyaar commented on modules/visual_script/doc_classes/VisualScriptMathConstant.xml:
```diff
@@ -4,7 +4,7 @@
 	</brief_description>
 	<description>
-		Provides common math constants, such as Pi or Euler's constant, on an output Data port.
+		Provides common math constants, such as Pi, on an output Data port.
```
This is okay. But could you let me know if the line 37-39 is also okay?
```
<constant name="MATH_CONSTANT_E" value="4" enum="MathConstant">
	Natural log: [code]2.718282[/code]
</constant>
```

willvincent replied:
There seem to be several similar terms around _e_, I'm honestly not sure whether that part needs to be reworded or not. Not sure how that should be phrased.. maybe just "Mathematical constant _e_"?

aaronfranke replied:
A quick Internet search for "Mathematical constant e" provides plenty of relevant information. I believe this phrasing is plenty useful and informative. We are to assume that Godot users have basic "Google-fu", after all.

@Noshyaar "Natural log: [code]2.718282[/code]" Well, natural log is a function, while E is a number. I would call it "Natural log base" or "Natural exponential base".

willvincent committed:
Remove incorrect & potentially confusing references to Euler

e is referred to as Euler’s number, so technically the MATH_EXP description in VisualScript doc was not incorrect, though could potentially lead to confusion.

e is different from Euler’s constant however, making the existing GDScript exp & VisualScriptMathConstant descriptions nvalid.

willvincent commented on modules/visual_script/doc_classes/VisualScriptMathConstant.xml:
```diff
@@ -35,7 +35,7 @@
 		</constant>
 		<constant name="MATH_CONSTANT_E" value="4" enum="MathConstant">
-			Natural log: [code]2.718282[/code]
+			Mathematical constant [code]e[/code], the natural log base: [code]2.718282[/code]
```
Better?

hpvb commented:
Cherry picked into 3.0.3
